### ISSUE TITLE
Sampler State always shows as POINT in the descriptor heap view

### ISSUE THREAD

**Issue by redorav** [baldurk/renderdoc#2528]
## Description

Sampler states in D3D12 always show up as POINT in the descriptor heap view. In this example, they should both be anisotropic.

![image](https://user-images.githubusercontent.com/700032/159597110-d6b80607-97b5-40ed-a67c-2368d505a856.png)

## Steps to reproduce

1. Create a descriptor heap
2. Create a few samplers in it
3. Inspect in renderdoc

## Environment

* RenderDoc version: 1.18 and Latest Nightly
* Operating System: Windows 11
* Graphics API: D3D12

**Comment by baldurk** [comment]
There is no such thing as a "descriptor heap view". Your screenshot is from the resource initialisation parameters from the resource inspector, are you sure that you understand that you are looking at the frame initial snapshot of the descriptor heap from inside an initialisation call which is explicitly marked as **internal**?

Generally speaking unless you are debugging RenderDoc there should be no reason to look at this internal initialisation. If you want to see which samplers are bound at any given action you can use the [pipeline state viewer](https://renderdoc.org/docs/window/pipeline_state.html) which shows you the used samplers and their parameters/configuration. This panel is designed to show you exactly this information.

I can trivially show that your steps to reproduce do _not_ fully show how to reproduce the problem by capturing one of the autotest nightly demos, which does exactly what your repro steps describe: https://github.com/baldurk/renderdoc/blob/ab228c51b8a35f703d526ef4777035721801fcad/util/test/demos/d3d12/d3d12_descriptor_indexing.cpp#L386-L393

The initial contents show samplers created with linear filters as I would expect, and if I modify it to create them as anisotropic that also shows up.

For the 'steps to reproduce' when reporting any bug you should provide _specific_ steps which anyone should be able to follow to reproduce the problem. If you believe that it is a general and easily reproduced problem then you should be able to point to an existing open source demo or tutorial, or make a minor modification to one, to demonstrate the issue. Unless you are positive you know exactly what the bug is, it may be that the problem you're seeing has a source other than what you think and can only be reproduced with more specific steps.

Listing steps like yours can easily be insufficient then as you omit the true source of the problem and leads to a bug report that cannot be investigated. On the other hand if you do provide a reproducible test case but the problem _is_ general enough to be described briefly, then it does no harm at all to show a specific instance of that problem.

**Comment by redorav** [comment]
Hi @baldurk it is possible that I am misunderstanding something.

I am looking at that view because I'm implementing my own D3D12 toy renderer to learn after having implemented Vulkan. My understanding is that what I've called the heap viewer is the state of the heap before the frame capture. I'm trying to see my samplers to see if they are properly initialized and I noticed this pattern where all other properties are propagated correctly except for the filter type. These samplers aren't used in any drawcall yet so I cannot use the pipeline viewer, but I wanted to see the heap to see what I might be doing wrong. If this is meant to be internal data, don't show it in the UI, but I think it has a ton of debugging value.

I can tell you that all I'm doing is described in those steps. Whether the actual repro steps may be more complex, I don't know, but I literally initialize a device, create a few resources and have a black screen with drawcalls that don't bind any resources. I'll try to get an executable to you for testing.
